src/current/advisories/a99561.md

```diff
@@ -0,0 +1,51 @@
+---
+title: Technical Advisory 99561
+advisory: A-99561
+summary: When adding a new column using the new declarative schema changer, CockroachDB retrieves a unique secondary index mistakenly, assuming it to be a primary index.
+toc: true
+affected_versions: v22.2.x, v23.1.0-v23.1.5
+advisory_date: 2023-07-10
+docs_area: releases
+---
+
+Publication date: {{ page.advisory_date | date: "%B %e, %Y" }}
+
+## Description
+
+A primary [index](../v23.1/indexes.html) is automatically created for each table, which indexes the table's [primary key](../v23.1/primary-key.html), or another unique value when there is no primary key. In CockroachDB v20.2 and v21.1 (the first two versions to support [`ALTER PRIMARY KEY`](../v21.1/alter-primary-key.html)), there was a bug where in some cases the index, which is automatically created to provide uniqueness for the existing primary key, would have its `EncodingType` field marked as `PrimaryIndexEncoding`. This behavior was fixed in [#66599](https://github.com/cockroachdb/cockroach/pull/66599), but left [secondary indexes](../v23.1/schema-design-indexes.html) with primary encoding.
```

Comment:
```suggestion
A primary [index](../v23.1/indexes.html) is automatically created for each table, which indexes the table's [primary key](../v23.1/primary-key.html), or another unique value when there is no primary key. In CockroachDB v20.2 and v21.1 (the first two versions to support [`ALTER PRIMARY KEY`](../v21.1/alter-primary-key.html)), there was a bug where in some cases the automatically-created index's `EncodingType` field marked as `PrimaryIndexEncoding`. This behavior was fixed in [#66599](https://github.com/cockroachdb/cockroach/pull/66599), but left [secondary indexes](../v23.1/schema-design-indexes.html) with primary encoding.
```
After adding the first sentence, does this suggestion make sense? I'm confused now because it seems like the bug is not in the primary index but in secondary indexes that are claiming to be primary indexes.